modules/lvgl/lvgl.c

```diff
@@ -21,58 +22,86 @@
+#define BUFFER_SIZE(n)                                                                             \
 	(CONFIG_LV_Z_BITS_PER_PIXEL *                                                              \
-	 ((CONFIG_LV_Z_VDB_SIZE * DISPLAY_WIDTH * DISPLAY_HEIGHT) / 100) / 8)
+	 ((CONFIG_LV_Z_VDB_SIZE * DISPLAY_WIDTH(n) * DISPLAY_HEIGHT(n)) / 100) / 8)
 #endif /* IS_MONOCHROME_DISPLAY */
 
-/* NOTE: depending on chosen color depth buffer may be accessed using uint8_t *,
- * uint16_t * or uint32_t *, therefore buffer needs to be aligned accordingly to
- * prevent unaligned memory accesses.
- */
-static uint8_t buf0[BUFFER_SIZE]
-#ifdef CONFIG_LV_Z_VDB_CUSTOM_SECTION
-	Z_GENERIC_SECTION(.lvgl_buf)
-#endif
-		__aligned(CONFIG_LV_Z_VDB_ALIGN);
+static uint32_t disp_buf_size[DT_ZEPHYR_DISPLAYS_COUNT] = {0};
+static uint8_t *buf0_p[DT_ZEPHYR_DISPLAYS_COUNT] = {NULL};
 
 #ifdef CONFIG_LV_Z_DOUBLE_VDB
-static uint8_t buf1[BUFFER_SIZE]
-#ifdef CONFIG_LV_Z_VDB_CUSTOM_SECTION
-	Z_GENERIC_SECTION(.lvgl_buf)
+static uint8_t *buf1_p[DT_ZEPHYR_DISPLAYS_COUNT] = {NULL};
 #endif
-		__aligned(CONFIG_LV_Z_VDB_ALIGN);
-#endif /* CONFIG_LV_Z_DOUBLE_VDB */
 
 #if ALLOC_MONOCHROME_CONV_BUFFER
-static uint8_t mono_vtile_buf[BUFFER_SIZE]
-#ifdef CONFIG_LV_Z_VDB_CUSTOM_SECTION
-	Z_GENERIC_SECTION(.lvgl_buf)
+static uint8_t *mono_vtile_buf_p[DT_ZEPHYR_DISPLAYS_COUNT] = {NULL};
 #endif
-		__aligned(CONFIG_LV_Z_VDB_ALIGN);
-#endif /* ALLOC_MONOCHROME_CONV_BUFFER */
+
+/* NOTE: depending on chosen color depth, buffers may be accessed using uint8_t *,*/
+/* uint16_t * or uint32_t *, therefore buffer needs to be aligned accordingly to */
+/* prevent unaligned memory accesses. */
+
+/* clang-format off */
+#define LV_DISPLAY_MACRO(n)                                                                        \
+	static uint8_t buf0_##n[BUFFER_SIZE(n)]							   \
+	IF_ENABLED(CONFIG_LV_Z_VDB_CUSTOM_SECTION, (Z_GENERIC_SECTION(.lvgl_buf)))                 \
+						       __aligned(CONFIG_LV_Z_VDB_ALIGN);           \
+                                                                                                   \
+	IF_ENABLED(CONFIG_LV_Z_DOUBLE_VDB, (							   \
+	static uint8_t buf1_##n[BUFFER_SIZE(n)]							   \
+	IF_ENABLED(CONFIG_LV_Z_VDB_CUSTOM_SECTION, (Z_GENERIC_SECTION(.lvgl_buf)))		   \
+			__aligned(CONFIG_LV_Z_VDB_ALIGN);					   \
+	))                                            \
+                                                                                                   \
+	IF_ENABLED(ALLOC_MONOCHROME_CONV_BUFFER, (						   \
+	static uint8_t mono_vtile_buf_##n[BUFFER_SIZE(n)]					   \
+	IF_ENABLED(CONFIG_LV_Z_VDB_CUSTOM_SECTION, (Z_GENERIC_SECTION(.lvgl_buf)))		   \
+			__aligned(CONFIG_LV_Z_VDB_ALIGN);					   \
+	))
+
+FOR_EACH(LV_DISPLAY_MACRO, (), LV_DISPLAYS_IDX_LIST);
+
+#define BUFS_INIT_MACRO(n)                                                                         \
```

Comment:
Naming: No need to abbreviate BUFFER and also adding MACRO as a suffix is a bit strange.